python/lsst/meas/extensions/scarlet/deconvolveExposureTask.py

```diff
@@ -0,0 +1,301 @@
+            mask=coadd.mask,
+            variance=coadd.variance,
+            dtype=coadd.image.array.dtype,
+        )
+        exposure = afwImage.Exposure(
+            maskedImage=maskedImage,
+            exposureInfo=coadd.getInfo(),
+            dtype=coadd.image.array.dtype,
+        )
+        return pipeBase.Struct(deconvolved=exposure)
+
+    def _removeHighFrequencySignal(
+        self, coadd: afwImage.Exposure
+    ) -> tuple[np.ndarray, np.ndarray]:
+        """Remove high frequency signal from the image and PSF.
+
+        This is done by performing a wavelet decomposition of the image
+        and PSF, setting the high frequency wavelets to zero, and
+        reconstructing the image and PSF from the remaining wavelets.
+
+        Parameters
+        ----------
+        coadd : `afwImage.Exposure`
+            Coadd image to deconvolve
+
+        Returns
+        -------
+        image : `np.ndarray`
+            Low frequency image
+        psf : `np.ndarray`
+            Low frequency PSF
+        """
+        psf = coadd.getPsf().computeKernelImage(coadd.getBBox().getCenter()).array
+        # Build the wavelet coefficients
+        wavelets = scl.detect.get_wavelets(
+            coadd.image.array[None, :, :],
+            coadd.variance.array[None, :, :],
+            scales=self.config.waveletScales,
+            generation=self.config.waveletGeneration,
+        )
+        # Remove the high frequency wavelets.
+        # This has the effect of preventing high frequency noise
+        # from interfering with the detection of peak positions.
+        wavelets[0] = 0
+        # Reconstruct the image from the remaining wavelet coefficients
+        image = scl.wavelet.starlet_reconstruction(
+            wavelets[:, 0],
+            generation=self.config.waveletGeneration,
+        )
+        # Remove the high frequency wavelets from the PSF.
+        # This is necesary for the image and PSF to have the
+        # same frequency content.
+        psf_wavelets = scl.wavelet.multiband_starlet_transform(
+            psf[None, :, :],
+            scales=self.config.waveletScales,
+            generation=self.config.waveletGeneration,
+        )
+        psf_wavelets[0] = 0
+        psf = scl.wavelet.starlet_reconstruction(
+            psf_wavelets[:, 0],
+            generation=self.config.waveletGeneration,
+        )
+        return image, psf
+
+    def _buildObservation(self, coadd: afwImage.Exposure, band: str):
+        """Build a scarlet lite Observation from an Exposure."""
+        bands = (band,)
+        model_psf = scl.utils.integrated_circular_gaussian(sigma=0.8)
+
+        if self.config.useWavelets:
+            image, psf = self._removeHighFrequencySignal(coadd)
+        else:
+            image = coadd.image.array
+            psf = coadd.getPsf().computeKernelImage(coadd.getBBox().getCenter()).array
+        weights = np.ones_like(coadd.image.array)
+        badPixelMasks = ["SAT", "INTRP", "NO_DATA"]
```

Comment:
I'm not sure. These are the same flags that I ignore in ScarletDeblendTask. I think my worry at the time was that poorly interpolated values can make things worse, and because the algorithm does a convolution the neighboring pixels will help constrain any masked pixels, so I'd rather ignore every pixel whose value is called into question.

Reply:
Okay then, this is probably worth investigating for both this task and Scarlet deblending once we switch to cell coadds, if you don't already have a ticket for that.

Reply:
Created DM-47571